.github/workflows/build.yml

```diff
@@ -212,11 +216,60 @@ jobs:
       - name: Lint podspec
         run: bundle exec pod spec lint *.podspec --allow-warnings --verbose
 
+  check_version:
+    name: Check Version is Unreleased
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+      tag: ${{ steps.tag_name.outputs.tag }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Find Version
+        id: version
+        run: grep version gradle.properties >> $GITHUB_OUTPUT
+      - name: Generate Tag Name
+        id: tag_name
+        run: echo tag=$TAG_NAME >> $GITHUB_OUTPUT
+        env:
+          TAG_NAME: v${{ steps.current_version.outputs.version }}
+      - name: Check if version was already released
+        run: "! git ls-remote -t --exit-code origin $TAG_NAME"
+        env:
+          TAG_NAME: ${{ steps.tag_name.outputs.tag }}
+
+  tag_release:
+    name: Tag Release & Bump Version
+    runs-on: ubuntu-latest
+    if: github.event_name == 'workflow_dispatch' && inputs.triggerRelease && github.ref == 'refs/heads/main'
+    needs: [ build, check_metadata, ktlint, lint, android_tests, ios_tests, js_tests, check_version ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Create Tag
+        uses: actions-ecosystem/action-push-tag@v1
+        with:
+          tag: ${{ needs.check_version.outputs.tag }}
+      - name: Generate next version
+        id: bump_version
+        uses: actions-ecosystem/action-bump-semver@v1
+        with:
+          current_version: ${{ needs.check_version.outputs.version }}
+          level: patch
+      - name: Update version for new development
+        run: sed -i "s/version=${{ needs.check_version.outputs.version }}/version=${{ steps.bump_version.outputs.new_version }}/" gradle.properties
+      - name: Create Version Bump Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          branch: versionBump
```

Comment:
I'm going to incorporate the new version number into the branch name so there won't be conflicts when pushing the branch